redis_builder.go

```diff
@@ -42,7 +42,7 @@ func DataStore() *DataStoreBuilder {
 // actual data store; that will be done by the SDK.
 type DataStoreBuilder struct {
 	prefix      string
-	pool        *r.Pool
+	pool        Pool
```

Comment:
Since the parameter type of `DataStoreBuilder.Pool` is still `*r.Pool`, you haven't really gained anything by this change— you will still only be able to pass a `*r.Pool` into the builder, even though it is being stored as the interface type `Pool`. That would've become apparent if you had tried to use this change or write a test for it.

However, for strict backward compatibility we cannot change the signature of the existing `Pool` method— even though _most_ code written against it would still work (since `*r.Pool` can always be cast to `Pool`), Go's type behavior regarding functions means that you could have some code that treats the method `DataStoreBuilder.Pool` as the specific type `func(*r.Pool) *DataStoreBuilder` and that code would break if the parameter were changed. So the right way to do this is to add a new method like `func (b *DataStoreBuilder) PoolInterface(Pool) *DataStoreBuilder`.

Reply:
Thanks for reviewing the change @eli-darkly , I'm sorry about missing tests. I thought it was a straightforward change. Added them now and followed your suggestion of adding a new method. 👍

Reply:
Well, it was a straightforward change, it was just done incompletely so you would not have been able to actually _use_ your change.

Reply:
🤦 Indeed.